src/main/java/de/eddyson/tapestry/react/services/nashorn/NashornModuleLoader.java

```diff
@@ -0,0 +1,70 @@
+package de.eddyson.tapestry.react.services.nashorn;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import javax.script.Bindings;
+import javax.script.ScriptEngine;
+import javax.script.ScriptException;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.tapestry5.internal.services.assets.ResourceChangeTracker;
+import org.apache.tapestry5.ioc.Resource;
+import org.apache.tapestry5.services.AssetSource;
+import org.apache.tapestry5.services.assets.StreamableResource;
+import org.apache.tapestry5.services.assets.StreamableResourceProcessing;
+import org.apache.tapestry5.services.assets.StreamableResourceSource;
+import org.apache.tapestry5.services.javascript.ModuleManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Nashorn module loader get's inserted into JavaScript executed by Nashorn and
+ * loads AMD modules from classpath
+ *
+ * @author Felix Gonschorek
+ */
+public class NashornModuleLoader {
+
+  private static final Logger log = LoggerFactory.getLogger(NashornModuleLoader.class);
+
+  private final ScriptEngine engine;
+  private final ModuleManager moduleManager;
+  private final Bindings bindings;
+  private final StreamableResourceSource srs;
+  private final ResourceChangeTracker tracker;
+
+  private final AssetSource assetSource;
+
+  public NashornModuleLoader(ScriptEngine engine, ModuleManager moduleManager, Bindings bindings,
+      StreamableResourceSource srs, ResourceChangeTracker tracker, AssetSource assetSource) {
+    super();
+    this.engine = engine;
+    this.moduleManager = moduleManager;
+    this.bindings = bindings;
+    this.srs = srs;
+    this.tracker = tracker;
+    this.assetSource = assetSource;
+  }
+
+  public void loadModule(String name) throws IOException, ScriptException {
+
+    Resource resource = null;
+    // special case "underscore" - TODO: investigate later
+    if ("underscore".equals(name)) {
+      resource = this.assetSource.getClasspathAsset("/META-INF/assets/tapestry5/underscore-1.8.3.js").getResource();
```

Comment:
In one of my applications, I override underscore with a newer version, that's probably going to cause trouble.
I assume, you need a special case because Tapestry uses a shim for underscore? Maybe it works by checking for "underscore-library" and using "underscore" as a fallback?